ethereum/messages.py

```diff
@@ -227,23 +234,14 @@ def _apply_casper_vote_transaction(state, tx):
 
     log_tx.debug("TX APPLIED", result=result, gas_remained=gas_remained,
                  data=data)
-
-    gas_used = tx.startgas - gas_remained
-
     # Transaction failed
     if not result:
         log_tx.debug('TX FAILED', reason='out of gas',
                      startgas=tx.startgas, gas_remained=gas_remained)
-        state.delta_balance(tx.sender, tx.gasprice * gas_remained)
-        state.delta_balance(state.block_coinbase, tx.gasprice * gas_used)
-        state.gas_used += gas_used
-        output = b''
-        success = 0
+        raise InvalidCasperVote("Failed Casper vote is considered an invalid tx")
```

Comment:
In the case of `initialize_epoch` failing, this error message will be confusing. I think we should just return `False, ""` and throw the error one level higher. Or we should discern what type of transaction here and throw accordingly.